cookies/views.py

```diff
@@ -1121,10 +1121,18 @@ def create_collection(request):
     """
     context = RequestContext(request, {})
 
+    parent_id = request.GET.get('parent_collection', None)
     template = loader.get_template('create_collection.html')
 
     if request.method == 'GET':
         form = UserAddCollectionForm()
+        if parent_id:
+            try:
+                parent_collection = _get_collection_by_id(request, int(parent_id))
+                authorization.check_authorization('change_collection', request.user, parent_collection)
+            except RuntimeError:
```

Comment:
check_authorization returns a boolean, rather than raising an exception; so this will not prevent an unauthorized user from adding a subcollection